### PR TITLE
Fix theming with SSR with CSRF safe approach

### PR DESCRIPTION
**Pull request by siddharthkp**
Following up on https://github.com/primer/react/pull/1868, the approach I chose relied on injecting an inline script to set a variable. This doesn't work in [github/docs-internal](https://github.com/github/docs-internal/pull/24123) because secure applications prevent executing inline scripts injected in the DOM 😇 

Inspired by `__NEXT_DATA__` from next.js, I've changed the implementation to inject a `<script type="application/json">` instead of `<script type="text/javascript">`.

[Tested with docs-internal](https://github.com/github/docs-internal/pull/24123/commits/625e6b4f6268e39e836375ac494a3f26f2c89dce#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R16), it works!

### CODE REVIEW

File: src/ThemeProvider.tsx [hunk 1]
```diff
@@ -37,6 +37,18 @@ const ThemeContext = React.createContext<{
   setNightScheme: () => null
 })
 
+// inspired from __NEXT_DATA__, we use application/json to avoid CSRF policy with inline scripts
+const getSeverHandoff = () => {
```
Comment (pksjce) [inline src/ThemeProvider.tsx:41]: typo :)

Comment (siddharthkp) [reply src/ThemeProvider.tsx]: loooool, fixed!

File: src/ThemeProvider.tsx [hunk 2]
```diff
@@ -37,6 +37,18 @@ const ThemeContext = React.createContext<{
...
 
+// inspired from __NEXT_DATA__, we use application/json to avoid CSRF policy with inline scripts
+const getServerHandoff = () => {
+  if (typeof document !== 'undefined' && document.getElementById('__PRIMER_DATA__')?.textContent) {
+    try {
+      return JSON.parse(document.getElementById('__PRIMER_DATA__')?.textContent as string)
+    } catch (error) {
+      // if JSON is invalid, supress error
+    }
+  }
```
Comment (colebemis) [inline src/ThemeProvider.tsx:48]: A possible way to clean up this code:

```suggestion
  try {
    const serverData = document?.getElementById('__PRIMER_DATA__')?.textContent
    if (serverData) {
      return JSON.parse(serverData)
    } 
  } catch (error) {
      // if JSON is invalid, supress error
  }
```

**Review by colebemis — APPROVED** [review]
Looks good! This is a clever trick 😅

File: src/ThemeProvider.tsx (hunk 2 above)
Comment (siddharthkp) [reply src/ThemeProvider.tsx]: I like it a lot!

**Pull request merged**